### ISSUE TITLE
bullseye does not like files with # in the name

### ISSUE THREAD

**Issue by wsnoble** [crux-toolkit/crux-toolkit#546]
If I run bullseye on a file with # in the name, it dies with the error message `invalid output file`.  If I rename the file to eliminate the #, then bullseye runs to completion.

Here is a brief demo:

```
(base) bash-4.2$ ~/proj/crux/crux-toolkit/build/src/crux bullseye --overwrite T sample#.mzML sample#.mzML 
INFO: Writing results to output directory 'crux-output'.
INFO: CPU: pyrrolysine.gs.washington.edu
INFO: Crux version: 4.1-eb571643-2021-12-17
INFO: Mon Dec 20 12:50:58 PST 2021
INFO: Beginning bullseye.
INFO: Hardklor v2.19, April 10 2015
INFO: Mike Hoopmann, Mike MacCoss
INFO: Copyright 2007-2015
INFO: University of Washington
Invalid output file.
(base) bash-4.2$ cp sample\#.mzML sample.mzML 
(base) bash-4.2$ ~/proj/crux/crux-toolkit/build/src/crux bullseye --overwrite T sample.mzML sample.mzML 
WARNING: The output directory 'crux-output' already exists.
Existing files will be overwritten.
WARNING: The file 'crux-output/bullseye.log.txt' already exists and will be overwritten.
INFO: CPU: pyrrolysine.gs.washington.edu
INFO: Crux version: 4.1-eb571643-2021-12-17
INFO: Mon Dec 20 12:51:19 PST 2021
INFO: Beginning bullseye.
INFO: Hardklor v2.19, April 10 2015
INFO: Mike Hoopmann, Mike MacCoss
INFO: Copyright 2007-2015
INFO: University of Washington
INFO: Reading from file: sample.mzML
INFO: Writing to file: crux-output/hardklor.mono.txt
INFO: 100
INFO: 
INFO: FastFewestPeptides analysis:
INFO:   Total number of scans analyzed: 2599
INFO: 
INFO: File access time: 0 minutes, 3 seconds.
INFO: Split Time:       0 minutes, 0 seconds.
INFO: Analysis Time:    0 minutes, 0 seconds.
INFO: IMPOSSIBLE!!!
INFO: Bullseye, v1.30, Apr 20, 2011
INFO: Copyright 2008-2011 Mike Hoopmann, Ed Hsieh, Mike MacCoss
INFO: University of Washington
INFO: 1 peptides from 2598 scans.
INFO: Finding persistent peptide signals:
0
INFO: No analysis results, exiting...
(base) bash-4.2$ 
```

**Comment by wsnoble** [comment]
Charles, can you check that this is still the case, and if so contact Hoopmann about it?

**Comment by CharlesEGrant** [comment]
This bug is still present. It's actually a bug in Hardklor, which is being run as part of bullseye. The underlying issue is that the arguments on the hardklor command line can be both commands setting the configuration for the hardklor run, and only the last two arguments can be filenames. The same parser function is used for both, and within that parser function '#' is used as a token delimiter, which ends up clobbering the filename when the parser attempts to split it into tokens.  I've dropped Mike an email asking if he would be able to fix this.

**Comment by CharlesEGrant** [comment]
Mike provided fixed code and I incorporated it in commit 87cb9f4405
